File: qna/src/main/java/com/example/rag/qna/domain/Question.java
package com.example.rag.qna.domain;

import core.rag.QuestionDTO;

import java.time.LocalDateTime;

public record Question(
        String userId,
        String questionId,
        String title,
        String category,
        String contents,
        LocalDateTime createdAt,
        LocalDateTime updatedAt) implements QuestionDTO {
}

File: rag-with-event-sourcing/src/main/kotlin/com/example/rag/qna/model/Question.kt
package com.example.rag.qna.model

import java.time.LocalDateTime
import java.util.UUID

data class Question(
    val id: UUID,
    val category: String,
    val title: String,
    val content: String,
    val userId: String,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null
)